src/main/java/org/bookkepervisualmanager/api/listeners/ContextInitializer.java

```diff
@@ -57,4 +67,42 @@ public void contextDestroyed(ServletContextEvent sce) {
         }
     }
 
+    /**
+     * Creates a {@link ConfigurationStore} following this priority order:
+     * <ol>
+     *  <li>Simple: System Property bookkeeper.visual.manager.serviceMetadataUri</li>
+     *  <li>Advanced: System Property bookkeeper.visual.manager.config.path</li>
+     *  <li>Advanced: Environment variable BVM_CONF_PATH</li>
+     *  <li>Advanced: web.xml property bookkeeper.visual.manager.config.path</li>
+     * </ol>
+     * @param context The Servlet context
+     * @return A {@link ConfigurationStore} containing the Bookkeeper configuration
+     * @throws ConfigurationNotValidException 
+     */
+    public ConfigurationStore buildInitialConfiguation(ServletContext context) throws ConfigurationNotValidException {
+        PropertyConfigurationResolver resolver;
```

Comment:
this block is the core of this stuff.

Your goal is to create a BookKeeper ClientConfiguration object.
There is no need of too much abstractions
```

ClientConfiguration conf = null;
String configPath = application.getAttribute("bookkeeper.visualmanager.configuration.path");
if (configPath == null) {
     ... read web xml configuration ?? (is it really useful?? maybe we can skip)
if (configPath == null) {
    configPath = System.getProperty("bookkeeper.visualmanager.configuration.path", null);
}
if (configPath != null) {
    conf = readConfigFromPath(configPath);
   return;
}
```
...otherwise try to look for a system property -DmetadataServiceUri=...... and create a ClientConfiguration for it
falling back to zk+null://localhost:2181/legders (or whatever is the BK default)